addons/account/models/account_journal.py

```diff
@@ -104,25 +109,30 @@ def _default_alias_domain(self):
 
     inbound_payment_method_ids = fields.Many2many('account.payment.method', 'account_journal_inbound_payment_method_rel', 'journal_id', 'inbound_payment_method',
-        domain=[('payment_type', '=', 'inbound')], string='For Incoming Payments', default=lambda self: self._default_inbound_payment_methods(),
+        domain=[('payment_type', '=', 'inbound')], string='Inbound Payment Methods', default=lambda self: self._default_inbound_payment_methods(),
         help="Manual: Get paid by cash, check or any other method outside of Odoo.\n"\
              "Electronic: Get paid automatically through a payment acquirer by requesting a transaction on a card saved by the customer when buying or subscribing online (payment token).\n"\
              "Batch Deposit: Encase several customer checks at once by generating a batch deposit to submit to your bank. When encoding the bank statement in Odoo,you are suggested to reconcile the transaction with the batch deposit. Enable this option from the settings.")
     outbound_payment_method_ids = fields.Many2many('account.payment.method', 'account_journal_outbound_payment_method_rel', 'journal_id', 'outbound_payment_method',
-        domain=[('payment_type', '=', 'outbound')], string='For Outgoing Payments', default=lambda self: self._default_outbound_payment_methods(),
+        domain=[('payment_type', '=', 'outbound')], string='Outbound Payment Methods', default=lambda self: self._default_outbound_payment_methods(),
         help="Manual:Pay bill by cash or any other method outside of Odoo.\n"\
              "Check:Pay bill by check and print it from Odoo.\n"\
              "SEPA Credit Transfer: Pay bill from a SEPA Credit Transfer file you submit to your bank. Enable this option from the settings.")
     at_least_one_inbound = fields.Boolean(compute='_methods_compute', store=True)
     at_least_one_outbound = fields.Boolean(compute='_methods_compute', store=True)
     profit_account_id = fields.Many2one('account.account',
         string='Profit Account',
-        domain="[('company_id', '=', company_id)]",
+        domain=lambda self: [('deprecated', '=', False), ('company_id', '=', 'company_id'),
+                             ('user_type_id.type', 'not in', ('receivable', 'payable')),
+                             ('user_type_id', 'in', (self.env.ref('account.data_account_type_revenue').id,
+                                                     self.env.ref('account.data_account_type_other_income').id))],
         check_company=True,
         help="Used to register a profit when the ending balance of a cash register differs from what the system computes")
     loss_account_id = fields.Many2one('account.account',
         string='Loss Account',
-        domain="[('company_id', '=', company_id)]",
+        domain=lambda self: [('deprecated', '=', False), ('company_id', '=', 'company_id'),
```

Comment:
fix typo: `('company_id', '=', 'company_id')` must be `('company_id', '=', company_id)`

Reply:
Also, I don't understand why the company is required on these fields but not on the accounts above. Is this additional domain still necessary or should we add this clause to others?